apps/quality_assurance/schema.py

```diff
@@ -0,0 +1,85 @@
+import graphene
+
+from graphene_django import DjangoObjectType
+
+from deep.permissions import ProjectPermissions as PP
+from lead.models import Lead
+
+from quality_assurance.models import EntryReviewComment, EntryReviewCommentText
+from .enums import CommentTypeEnum
+
+
+def get_entry_comment_qs(info):
+    """
+    NOTE: To be used in EntryReviewCommentDetailType
+    """
+    entry_comment_qs = EntryReviewComment.objects.filter(
+        entry__project=info.context.active_project
+    )
+    # Generate queryset according to permission
+    if PP.check_permission(info, PP.Permission.VIEW_ENTRY):
+        if PP.check_permission(info, PP.Permission.VIEW_ALL_LEAD):
+            return entry_comment_qs
+        elif PP.check_permission(info, PP.Permission.VIEW_ONLY_UNPROTECTED_LEAD):
+            return entry_comment_qs.filter(entry__lead__confidentiality=Lead.Confidentiality.UNPROTECTED)
+    return EntryReviewComment.objects.none()
+
+
+class EntryReviewCommentTextType(DjangoObjectType):
+    class Meta:
+        model = EntryReviewCommentText
+        field = (
+            'text',
+            'created_at',
+            'id'
+        )
+
+
+class EntryReviewCommentType(DjangoObjectType):
+    class Meta:
+        model = EntryReviewComment
+        fields = (
+            'id',
+            'entry',
+            'created_by',
+            'created_at',
+            'mentioned_users'
+        )
+
+    comment_type = graphene.Field(CommentTypeEnum)
+    text = graphene.Field(graphene.String)
+
+    @staticmethod
+    def resolve_text(root, info):
+        return root.annotated_text  # This is resolved from ReviewCommentsLoader
+
+
+class EntryReviewCommentDetailType(EntryReviewCommentType):
+    class Meta:
+        model = EntryReviewComment
+        skip_registry = True
+        fields = (
+            'id',
+            'entry',
+            'created_by',
+            'created_at',
+            'mentioned_users'
+        )
+
+    text_history = graphene.List(graphene.NonNull(EntryReviewCommentTextType))
+
+    @staticmethod
+    def get_custom_queryset(queryset, info, **kwargs):
+        return get_entry_comment_qs(info)
+
+    @staticmethod
+    def resolve_text_history(root, info, **kwargs):
+        return info.context.dl.quality_assurance.text_history.load(root.pk)
+
+
+class Query:
+    review_comments = graphene.List(EntryReviewCommentDetailType)
```

Comment:
This should be single. 
```suggestion
    review_comment = DjangoObjectField((EntryReviewCommentDetailType)
```